robot/impl/resource_manager.go

```diff
@@ -1019,14 +1026,37 @@ func (manager *resourceManager) markResourcesRemoved(
 func (manager *resourceManager) removeOrphanedResources(ctx context.Context,
 	rNames []resource.Name,
 ) {
-	// Ignore returned resources to close, as removeMarkedAndClose will handle.
 	manager.markResourcesRemoved(rNames, nil)
-	if _, err := manager.removeMarkedAndClose(ctx, nil); err != nil {
+	if err := manager.removeMarkedAndClose(ctx, nil); err != nil {
 		manager.logger.Errorw("error removing and closing marked resources",
 			"error", err)
 	}
 }
 
+// createConfig will create a config.Config based on the current state of the
+// resource graph, processManager and moduleManager.
+func (manager *resourceManager) createConfig() *config.Config {
+	conf := &config.Config{}
```

Comment:
`createConfig` used to start with `manager.configLock.Lock()`, but since `completeConfig` acquires that lock and tries to `createConfig` itself (both for weak dependencies and the framesystem), that caused deadlocking. I still think this method should be guarded somehow; looking into it.

Reply:
On second thought, I'm not sure we need a lock here. 

Our current, internal usages of `localRobot.Config()` will never overlap—there's one in `Reconfigure`, one in `updateWeakDependents`, and two in framesystem reconfiguration functions—these usages currently always happen sequentially and in that order. If we add more usages or a user has asynchronous overlapping usages, not locking here only means that `localRobot.Config()` will return the state of the resource graph, process manager and module manager at exactly the time the method was called. Underlying calls to the resource graph are locked anyway, so there should be no reading/writing at the same time.